drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -239,17 +279,26 @@ static int sdw_get_group_count(struct sdw_bus *bus,
 	if (!group->rates)
 		return -ENOMEM;
 
+	group->lanes = kcalloc(group->max_size, sizeof(int), GFP_KERNEL);
+	if (!group->lanes)
+		return -ENOMEM;
```

Comment:
you need to free `group->rates` - and set it to NULL to be safe.

Reply:
> you need to free `group->rates` - and set it to NULL to be safe.

done